File: commands/music/controller.js
/* global controller client guild include */

// '⏮'

const Controller = include( 'utils/controller' );

module.exports = {
  name: 'controller',
  description: 'Skickar musik kontrollen.',
  longDescription: 'Skickar en kontroll vilket man kan kontrollera musiken med.',
  aliases: [ 'kontroll' ],
  group: 'music',
  usage: 'controller',
  example: 'controller',
  serverOnly: true,
  adminOnly: false,
  async execute( message, args ) {
    if ( !controller ) global.controller = new Controller( client, guild );
    else controller.newController();
  },
};

File: commands/music/controller.go
package music

import (
	"github.com/digitalungdom-se/dub/internal"
	"github.com/digitalungdom-se/dub/pkg"
)

var Controller = internal.Command{
	Name:        "controller",
	Description: "Skickar en ny controller i musik kanalen",
	Aliases:     []string{"kontroll"},
	Group:       "music",
	Usage:       "controller",
	Example:     "controller",
	ServerOnly:  true,
	AdminOnly:   false,

	Execute: func(ctx *pkg.Context, server *internal.Server) error {
		ctx.Delete()
		err := server.Controller.NewControllerMessage(&server.ReactionListener)

		return err
	},
}
